### PR TITLE
[FEATURE] Add new `runCommand` step

### PR DESCRIPTION
**Pull request by mteu**
## Content
This PR addresses the feature request https://github.com/CPS-IT/project-builder/issues/227 asking for a step that allows to run a console command in the process project directory.

* Identifier: **`runCommand`**
* Implementation: [`Builder\Generator\Step\RunCommandStep`](https://github.com/CPS-IT/project-builder/blob/main/src/Builder/Generator/Step/RunCommandStep.php)

The `runCommand` enables you to execute custom shell commands in the generated project directory.
Since this steps requires a `BuildResult` it makes most sense to execute your command after dependent steps have
been executed. That can be either in the temporary or working (target) directory.

Please keep in mind that an already executed command cannot be reverted.

## Example
```yaml
// config.yaml

steps:
  - type: runCommand
    options:
      command: 'composer install'
```

## Pending Tasks:

- [x] Documentation
- [x] Unit Tests
- [x] Write command output to console

### CODE REVIEW

File: tests/src/Builder/Generator/Step/RunCommandStepTest.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+use PHPUnit\Framework;
+use Symfony\Component\Filesystem;
+
+/**
+ * RunCommandStepTest.
+ *
+ * @author Martin Adler <mteu@mailbox.org>
+ * @license GPL-3.0-or-later
+ */
+class RunCommandStepTest extends Tests\ContainerAwareTestCase
```
Comment (mteu) [inline tests/src/Builder/Generator/Step/RunCommandStepTest.php:37]:
```suggestion
final class RunCommandStepTest extends Tests\ContainerAwareTestCase
```

File: docs/development/build-steps.md [hunk 2]
```diff
@@ -178,6 +178,17 @@ step.
 Read more at [`Configuration#Shared source files`](configuration.md#shared-source-files).
 ```
 
+### Run Command
+
+* Identifier: **`runCommand`**
+* Implementation: [`Builder\Generator\Step\RunCommandStep`](https://github.com/CPS-IT/project-builder/blob/main/src/Builder/Generator/Step/RunCommandStep.php)
+
+The `runCommand` enables you to execute custom shell commands in the generated project directory.
+Since this steps requires a `BuildResult` it makes most sense to execute your command after dependent steps have
```
Comment (eliashaeussler) [inline docs/development/build-steps.md:187]: Every step requires a `BuildResult` which makes this sentence probably a bit misleading. Can you rephrase it?

**Review by eliashaeussler — CHANGES_REQUESTED** [review]

**Review by eliashaeussler — APPROVED** [review]
